From f5e3b43de3a5e3e6332106282bbcaf336e8204a5 Mon Sep 17 00:00:00 2001
From: Johannes Berg <johannes.berg@intel.com>
Date: Wed, 15 Mar 2023 16:08:22 +0100
Subject: [PATCH 13/28] util: don't print EHT info if not present

That's just confusing, don't print it if the EHT MAC
capabilities attribute isn't there (the kernel puts
all the attrs together.)

Signed-off-by: Johannes Berg <johannes.berg@intel.com>
---
 util.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/util.c b/util.c
index 80dc301..50e3fbc 100644
--- a/util.c
+++ b/util.c
@@ -1593,7 +1593,8 @@ void print_eht_info(struct nlattr *nl_iftype, int band)
 	nla_parse(tb, NL80211_BAND_IFTYPE_ATTR_MAX,
 		  nla_data(nl_iftype), nla_len(nl_iftype), NULL);
 
-	if (!tb[NL80211_BAND_IFTYPE_ATTR_IFTYPES])
+	if (!tb[NL80211_BAND_IFTYPE_ATTR_IFTYPES] ||
+	    !tb[NL80211_BAND_IFTYPE_ATTR_EHT_CAP_MAC])
 		return;
 
 	printf("\t\tEHT Iftypes: ");
-- 
2.39.2

